packages/mobile/src/services/offline-downloader/offline-storage.ts

```diff
@@ -170,15 +171,15 @@ export const listTracks = async (): Promise<string[]> => {
 
 export const getTrackJson = async (
   trackId: string
-): Promise<Track & UserTrackMetadata> => {
+): Promise<Nullable<Track & UserTrackMetadata>> => {
   try {
     const trackJson = await readFile(getLocalTrackJsonPath(trackId), 'utf8')
     return JSON.parse(trackJson)
   } catch (e) {
     if (e instanceof SyntaxError) {
       purgeDownloadedTrack(trackId)
     }
-    return Promise.reject(e)
+    return null
```

Comment:
Nice. Related to the "before" code - are so many unhandled promise rejections...

Reply:
yeah agreed, after delete is batched, im keen to ensure our logs are clean except for true errors